### PR TITLE
fix: encodeURI to correctly load specs with white space

### LINKED ISSUE #16278: Cypress component testing could not detect tests when spec file contains spaces in path
### Current behavior
Cypress component testing (cypress open-ct) could not detects tests when spec file contains spaces in path. When trying to load a spec in browser following message is displayed:

 No tests found.
Cypress could not detect tests in this file.

Example:
OS: Windows 10

path is: 

c:\projects\my project\cypress\component\test.spec.js

if I change the path to:

c:\projects\my_project\cypress\component\test.spec.js

then tests are successfully detected.

### Desired behavior
Tests would be detected

### Test code to reproduce
Just include space in path to spec file

### Versions

Cypress 7.2.0
OS: Windows 10

BTW: Cypress rules and keep up the good work :)

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://github.com/cypress-io/cypress/issues/16278

### User facing changelog

Support loading specs with spaces in the path.

### Additional details

Need to do encodeURI to match the specs correctly.

### How has the user experience changed?

Specs with white space now work (eg `C://Jon Doe/White Space.spec.tsx`).

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: npm/webpack-dev-server/src/loader.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ const makeImport = (file: Cypress.Cypress['spec'], filename: string, chunkName:
   const magicComments = chunkName ? `/* webpackChunkName: "${chunkName}" */` : ''
 
   return `"${filename}": {
-    shouldLoad: () => document.location.pathname.includes("${file.absolute}"),
+    shouldLoad: () => document.location.pathname.includes("${encodeURI(file.absolute)}"),
```
Comment (agg23) [inline npm/webpack-dev-server/src/loader.ts:20]: So the problem was that we weren't validating the URL correctly, but we were properly pushing the encoded URL?

**Review by agg23 — DISMISSED** [review]

**Review by elevatebart — COMMENTED** [review]
Have you tried this on windows?

File: packages/runner-ct/cypress/component/Spec With Whitespace.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { mount } from '@cypress/react'
+import RunnerCt from '../../src/app/RunnerCt'
+import '@packages/runner/src/main.scss'
+import { makeState, fakeConfig, FakeEventManager } from './utils'
```
Comment (elevatebart) [inline packages/runner-ct/cypress/component/Spec With Whitespace.spec.tsx:5]: This did not exist in utils.ts

File: packages/runner-ct/src/lib/state.ts [hunk 3]
```diff
@@ -293,7 +293,7 @@ export default class State {
   }
 
   @action updateSpecByUrl (specUrl) {
-    const foundSpec = _.find(this.specs, { name: specUrl })
+    const foundSpec = _.find(this.specs, { name: decodeURI(specUrl) })
```
Comment (elevatebart) [inline packages/runner-ct/src/lib/state.ts:296]: I added this to be able to resolve the active spec from the URL when there are spaces

**Review by elevatebart — APPROVED** [review]

File: npm/webpack-dev-server/src/loader.ts (hunk 1 above)
Comment (lmiller1990) [reply npm/webpack-dev-server/src/loader.ts:20]: that's right!

File: packages/runner-ct/src/lib/state.ts (hunk 3 above)
Comment (lmiller1990) [reply packages/runner-ct/src/lib/state.ts:296]: Thanks

Comment (lmiller1990) [reply packages/runner-ct/src/lib/state.ts:296]: good catch

**Pull request merged**